File: Java/OOP February - April 2021/02. Encapsulation/E05FootballTeamGenerator/Player.java
package E05FootballTeamGenerator;

public class Player {
    private String name;
    private int endurance;
    private int sprint;
    private int dribble;
    private int passing;
    private int shooting;

    public Player(String name, int endurance, int sprint, int dribble, int passing, int shooting) {
        this.setName(name);
        this.setEndurance(endurance);
        this.setSprint(sprint);
        this.setDribble(dribble);
        this.setPassing(passing);
        this.setShooting(shooting);
    }

    private void setName(String playerName) {
        if (playerName == null || playerName.trim().isEmpty()) {
            throw new IllegalArgumentException("A name should not be empty.");
        }
        this.name = playerName;
    }

    private void setEndurance(int endurance) {
        if (endurance < 0 || 100 < endurance) {
            throw new IllegalArgumentException(String.format("%s should be between 0 and 100.", "Endurance"));
        }
        this.endurance = endurance;
    }

    private void setSprint(int sprint) {
        if (sprint < 0 || 100 < sprint) {
            throw new IllegalArgumentException(String.format("%s should be between 0 and 100.", "Sprint"));
        }
        this.sprint = sprint;
    }

    private void setDribble(int dribble) {
        if (dribble < 0 || 100 < dribble) {
            throw new IllegalArgumentException(String.format("%s should be between 0 and 100.", "Dribble"));
        }
        this.dribble = dribble;
    }

    private void setPassing(int passing) {
        if (passing < 0 || 100 < passing) {
            throw new IllegalArgumentException(String.format("%s should be between 0 and 100.", "Passing"));
        }
        this.passing = passing;
    }

    private void setShooting(int shooting) {
        if (shooting < 0 || 100 < shooting) {
            throw new IllegalArgumentException(String.format("%s should be between 0 and 100.", "Shooting"));
        }
        this.shooting = shooting;
    }

    public String getName() {
        return this.name;
    }

    public int getEndurance() {
        return this.endurance;
    }

    public int getSprint() {
        return this.sprint;
    }

    public int getDribble() {
        return this.dribble;
    }

    public int getPassing() {
        return this.passing;
    }

    public int getShooting() {
        return this.shooting;
    }

    public double overallSkillLevel() {
        return 1.0 * (this.getEndurance() + this.getSprint()
                + this.getDribble() + this.getPassing()
                + this.getShooting()) / 5;
    }
}

File: C#/OOP/02_Encapsulation_Exercise/FootballTeamGenerator/Model/Player.cs
﻿using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace FootballTeamGenerator.Model
{
    public class Player
    {
        private string name;
        private int endurance;
        private int sprint;
        private int dribble;
        private int passing;
        private int shooting;

        public Player(string name, int endurance, int sprint, int dribble, int passing, int shooting)
        {
            Name = name;
            Endurance = endurance;
            Sprint = sprint;
            Dribble = dribble;
            Passing = passing;
            Shooting = shooting;
        }

        public string Name
        {
            get { return this.name; }
            private set
            {
                if (string.IsNullOrWhiteSpace(value))
                {
                    throw new ArgumentException("A name should not be empty.");
                }

                this.name = value;
            }
        }

        public int Endurance
        {
            get { return this.endurance; }
            private set
            {
                if (value < 1 || value > 100)
                {
                    throw new ArgumentException($"{nameof(Endurance)} should be between 0 and 100.");
                }

                this.endurance = value;
            }
        }
        public int Sprint
        {
            get { return this.sprint; }
            private set
            {
                if (value < 1 || value > 100)
                {
                    throw new ArgumentException($"{nameof(Sprint)} should be between 0 and 100.");
                }

                this.sprint = value;
            }
        }

        public int Dribble
        {
            get { return this.dribble; }
            private set
            {
                if (value < 1 || value > 100)
                {
                    throw new ArgumentException($"{nameof(Dribble)} should be between 0 and 100.");
                }

                this.dribble = value;
            }
        }

        public int Passing
        {
            get { return this.passing; }
            private set
            {
                if (value < 1 || value > 100)
                {
                    throw new ArgumentException($"{nameof(Passing)} should be between 0 and 100.");
                }

                this.passing = value;
            }
        }

        public int Shooting
        {
            get { return this.shooting; }
            private set
            {
                if (value < 1 || value > 100)
                {
                    throw new ArgumentException($"{nameof(Shooting)} should be between 0 and 100.");
                }

                this.shooting = value;
            }
        }

        public double Stats()
        {
            return (Endurance + Sprint + Dribble + Passing + Shooting) / 5.0;
        }


    }
}
